### ISSUE TITLE
gPodder losing almost all episodes in its list (downloaded files still exist)

### ISSUE THREAD

**Issue by DaPadrePedro** [gpodder/gpodder#1184]
Hi,

I have a few dozens of podcasts listed in gPodder, but from time to time it seems gPodder crashes and loses its episode list (or at least for most podcasts). The downloaded files are still there, but the episodes are not listed anymore in gPodder. This is now the 3rd time I've experienced this, meaning my gPodder episode list is almost relatively "empty" right now (only the episodes known in the RSS feeds show up), which is not a great experience of course... Instead of many thousands of episodes I now have only a few hundreds listed anymore... :-(

As I've just said, this is not the first time I've experienced this. This has occurred to me with different versions of the official gPodder. The log logically mentions a lot of "WARNING: Unknown external file" entries. Before that I must say I saw a lot of entries like this:
2021-10-27 10:20:08,737 [gpodder.dbsqlite] ERROR: Cannot save <gpodder.gtkui.model.GPodcast object at 0x06db8710>: database is locked
Traceback (most recent call last):
  File "C:\Program Files (x86)\gPodder\lib\python3.9\site-packages\gpodder\dbsqlite.py", line 220, in _save_object
    cur.execute(sql, values)
sqlite3.OperationalError: database is locked

Restarting gPodder didn't help and yes, I only had 1 gPodder process running. Restarting my system though did make this DB locked log entry disappear, so something must indeed have locked the DB.

Is there a way to get my episode list back?
And to prevent this from happening again?

Thank you!

**Comment by DaPadrePedro** [comment]
Hi,
Sorry for the late reply... I've executed the command (through DB Browser for SQLite) and just got an OK immediately.

Regards,
Pedro

**Comment by GoMeister** [comment]
I've been experiencing a similar issue. but it  looks like the new release 3.11.0 may have cleared it up.  Are there any tools to purge the duplicate episodes I have in the download folders?  They must be several GB by now.

**Comment by auouymous** [comment]
You could grep the log file for the unknown file warnings into sed to isolate the filename and pipe them into xargs to delete.

There is no way for gpodder to associate a file with an episode if the RSS feed has removed the episode, and all Youtube episodes lack a filename and can't be associated with files. I really want to add a feature to download an URL and create a fake episode in a fake channel, or any real channel. It would then be nice if gpodder could show a list of unknown files and let the user manually associate them with a channel, creating fake episodes for files removed from the feed. It might also be possible to match filename to episode title and automatically select the guessed channel and episode (although renaming extensions would be an issue). Fake episodes would obviously be missing some metadata such as description and thumbnail, but at least they would be back in gpodder.

**Comment by DaPadrePedro** [comment]
Yep, some tools/features to do some manual work per episode or even in batch would be great!

**Comment by johnraff** [comment]
> It would then be nice if gpodder could show a list of unknown files and let the user manually associate them with a channel, creating fake episodes for files removed from the feed.

Having just returned to a long-neglected gpodder database, I can say how much that would be appreciated! I've got dozens of downloaded, but now "orphaned" episodes that are no longer on the RSS feeds. Of course they can be played directly from the Downloads directory, but it would be nice if gpodder could be configured to recognize them.
